src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,332 @@
+      setListItems(newList)
+    }
+    setCurrentInputResponses(resetCurrentResponses(inputFields))
+    setOpen(false)
+    setSelectedListItem(null)
+  }
+
+  const editItem = async (index: number) => {
+    setSelectedListItem(index)
+    setCurrentInputResponses(listItems[index])
+    setOpen(true)
+  }
+
+  const deleteItem = async (index: number) => {
+    setListItems(listItems.filter((_, i) => i !== index))
+    setOpen(false)
+    setCurrentInputResponses(resetCurrentResponses(inputFields))
+  }
+
+  const listDisplayProps: ListLayoutProps = {
+    listItems,
+    displayFormat,
+    editItem,
+    deleteItem,
+    fieldTitles: inputFields.map((e: TemplateElement) => e.title),
+    Markdown,
+  }
+
+  const DisplayComponent =
+    displayType === DisplayType.TABLE ? (
+      <ListTableLayout {...listDisplayProps} />
+    ) : (
+      <ListCardLayout {...listDisplayProps} />
+    )
+
+  return (
+    <>
+      <label>
+        <Markdown text={label} semanticComponent="noParagraph" />
+      </label>
+      <Markdown text={description} />
+      <Modal
+        size="tiny"
+        // closeIcon
+        onClose={() => {
+          setOpen(false)
+          setSelectedListItem(null)
+          setCurrentInputResponses(resetCurrentResponses(inputFields))
+          setInputError(false)
+        }}
+        onOpen={() => setOpen(true)}
+        open={open}
+        trigger={<Button primary content={createModalButtonText} />}
+      >
+        <Segment>
+          <Form>
+            <Markdown text={modalText} />
+            {inputFields.map((field: TemplateElement, index: number) => {
+              const element = {
+                code: field.code,
+                pluginCode: field.elementTypePluginCode as string,
+                category: field.category as TemplateElementCategory,
+                title: field.title as string,
+                parameters: field.parameters,
+                isEditable: true, // Update
+                isRequired: field.isRequired ?? true,
+                isVisible: true,
+                validationExpression: field?.validation || true,
+                validationMessage: field?.validationMessage || '',
+                id: index, // This is the only link between element and the response
+                elementIndex: 0,
+                page: 0,
+                sectionIndex: 0,
+                helpText: null,
+                sectionCode: '0',
+              }
+              return (
+                <ApplicationViewWrapper
+                  key={`list-${element.code}`}
+                  element={element}
+                  isStrictPage={inputError}
+                  allResponses={allResponses}
+                  currentResponse={
+                    selectedListItem === null
+                      ? {
+                          id: index,
+                          code: element.code,
+                          text: currentInputResponses?.[index]?.value.text,
```

Comment:
would the be a case where this has value ?

Reply:
No, but if there's no "text" field present, the inner item goes immediately to "strict" error mode, so shows validation errors immediately. I can't totally remember why this happens, but can confirm that if I set this to "" or `null`, you get immediate validation fail on loading the modal.

Reply:
Ah yes, that's right. This does have a value, as it keeps getting updated whenever a fields loses focus.